compiler/luci/service/src/CircleTypeInferenceRule.cpp

```diff
@@ -572,6 +577,13 @@ struct TypeInferenceAlgorithm final : public luci::CircleNodeVisitor<loco::DataT
     return then_graph_output->dtype();
   }
 
+  loco::DataType visit(const luci::CircleNonMaxSuppressionV4Out *node) final
+  {
+    (void)node;
+    assert(node->index() == 0 || node->index() == 1);
+    return loco::DataType::S32;
```

Comment:
@jinevening , is this OK for quantization?

Reply:
I think it is ok for quantization, because TFLite does not have a quantized kernel for this Op.

I have another question to @ai-moiseev .

TFLite kernel for NonMaxSuppression has three outputs: selected_indices, selected_scores, num_selected_indices. But this draft assumes it has only two outputs. Why did you assume there are only two outputs?

Reply:
@jinevening TFLite kernel has additional output ``selected_scores`` in case when ``is_soft_nms`` is true and the node gets marked as NonMaxSuppressionV5 in tflite model. Otherwise it's saved as NonMaxSuppressionV4 with ``selected_indices`` and ``num_selected_indices`` outputs.

Reply:
@ai-moiseev , I didn't know that. Thanks.